source/common/config/well_known_names.cc

```diff
@@ -183,7 +183,9 @@ TagNameValues::TagNameValues() {
   addRe2(RDS_ROUTE_CONFIG, R"(^http\.<TAG_VALUE>\.rds\.((<ROUTE_CONFIG_NAME>)\.)\w+?$)", ".rds.");
 
   // vhost.[<virtual host name>.]route.(<route_stat_prefix>.)*
-  addTokenized(ROUTE, "vhost.*.route.$.**");
+  // Note: <route_stat_prefix> can contain dots thus we have to maintain full
+  // match.
+  addRe2(ROUTE, R"(^vhost\.<TAG_VALUE>\.route\.((.*?)\.)\w+?$)", "vhost.");
```

Comment:
what is difference between using `ROUTE_CONFIG_NAME` as the regexp expansion similar to the `RDS_ROUTE_CONFIG`?

Reply:
can you use `addTokenized(ROUTE, "vhost.**.route.$.**")` ?

That would be faster than a regex.

Reply:
I gave `addTokenized(ROUTE, "vhost.**.route.$.**")` a try, it does not work as intended.